### DISCUSSION TITLE
Conditional return types reference another method

### DISCUSSION THREAD

**Discussion by Wulfheart** [vimeo/psalm#9816] [Q&A]
Given the following file 
```php
<?php
final class Sitemap
{
    /**
     * @var array <string, SitemapPage>
     */
    private array $pages = [];

    public function containsPage(PageId $pageId): bool
    {
        return array_key_exists($pageId->getAsString(), $this->pages);
    }

    public function getPage(PageId $pageId): ?SitemapPage
    {
        return $this->pages[$pageId->getAsString()] ?? null;
    }
}
```
I would like to somehow define that the `getPage` function is not null if the result of the `containsPage` is true. Just like the IDE recognises it with null checks. Is this somehow possible?

**Comment by orklah** [comment]
That should help: https://psalm.dev/docs/annotating_code/adding_assertions/#asserting-return-values-of-methods

**Reply by Wulfheart** [reply]
Thank you really much. However, it does not seem to work when using something like `@psalm-assert-if-true SitemapPage $this->getPage($pageId)`

**Reply by orklah** [reply]
oh sorry, I hadn't noticed that you had an argument in there. That's not supported for now
